homeassistant/components/sonos/switch.py

```diff
@@ -357,6 +360,43 @@ async def async_handle_switch_on_off(self, turn_on: bool) -> None:
             _LOGGER.error("Could not update %s: %s", self.entity_id, exc)
 
 
+@callback
+def async_migrate_alarm_unique_ids(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    household_id: str,
+    alarm_ids: list[str],
+) -> None:
+    """Migrate alarm switch unique_ids in the entity registry to the new format."""
+    entity_registry = er.async_get(hass)
+    registry_entries = er.async_entries_for_config_entry(
+        entity_registry, config_entry.entry_id
+    )
+
+    alarm_entries = [
+        (entry.unique_id, entry)
+        for entry in registry_entries
+        if entry.domain == Platform.SWITCH and entry.original_icon == "mdi:alarm"
```

Comment:
Is there a better way than checking the icon?

Reply:
That has been the default icon since the feature was added. This seemed easier than parsing the `entity_id` or `original_name` strings. (It's also why the new `unique_id` has "alarm" in the string.)

For reference, here's what's available in the entity registry for "old" alarms:
```json
            {
                "area_id": null,
                "capabilities": null,
                "config_entry_id": "c9145e60ca0bf47a27e168debc5b5854",
                "device_class": null,
                "device_id": "d6931d510a72b2ff5db5de54a7d021b8",
                "disabled_by": null,
                "entity_category": "config",
                "entity_id": "switch.sonos_alarm_7831",
                "icon": null,
                "id": "f6010779b6269e23884cf0116f4b5dfb",
                "name": null,
                "original_device_class": null,
                "original_icon": "mdi:alarm",
                "original_name": "Bedroom Daily Alarm 07:00",
                "platform": "sonos",
                "supported_features": 0,
                "unique_id": "sonos-7831",
                "unit_of_measurement": null
            }
```

Reply:
Its a bit ugly, but it should be deterministic in this case